plugins/kafka/src/index.ts

```diff
@@ -0,0 +1,221 @@
+        path: join(messageBrokerDirectory, generateFileName),
+      },
+    ];
+  }
+
+  beforeUpdateEnv(
+    context: DsgContext,
+    eventParams: CreateServerDotEnvParams["before"]
+  ): CreateServerDotEnvParams["before"] {
+    const resourceName = context.resourceInfo?.name;
+    if (!resourceName) {
+      throw new Error("Resource name is missing");
+    }
+    const vars = {
+      KAFKA_BROKERS: "localhost:9092",
+      KAFKA_ENABLE_SSL: "false",
+      KAFKA_CLIENT_ID: kebabCase(resourceName),
+      KAFKA_GROUP_ID: kebabCase(resourceName),
+    };
+    const newEnvParams = [
+      ...eventParams.envVariables,
+      ...Object.entries(vars).map(([key, value]) => ({ [key]: value })),
+    ];
+    return { envVariables: newEnvParams };
+  }
+
+  beforeUpdateJson(
+    context: DsgContext,
+    eventParams: CreatePackageJsonParams["before"]
+  ): CreatePackageJsonParams["before"] {
+    const myValues = {
+      dependencies: {
+        "@nestjs/microservices": "^8.2.3",
+        kafkajs: "^2.2.0",
+      },
+    };
+    const newValues = merge(eventParams.updateValues, myValues);
+
+    return { ...eventParams, updateValues: newValues };
+  }
+
+  async afterCreateMessageBrokerService(
+    context: DsgContext,
+    eventParams: CreateMessageBrokerServiceParams["after"]
+  ): Promise<Module[]> {
+    const { serverDirectories } = context;
+    const { messageBrokerDirectory } = serverDirectories;
+    const templatePath = resolve(staticDirectory, `kafka.service.template.ts`);
+
+    const template = await readFile(templatePath, "utf8");
+    const generateFileName = `kafka.service.ts`;
+
+    const path = join(messageBrokerDirectory, generateFileName);
+    return [{ code: template, path }];
+  }
+  async afterCreateMessageBrokerServiceBase(
+    context: DsgContext,
+    eventParams: CreateMessageBrokerServiceBaseParams["after"]
+  ): Promise<Module[]> {
+    const { serverDirectories } = context;
+    const { messageBrokerDirectory } = serverDirectories;
+    const templatePath = resolve(
+      staticDirectory,
+      `kafka.service.base.template.ts`
+    );
+
+    const template = await readFile(templatePath, "utf8");
+    const generateFileName = `kafka.service.base.ts`;
+
+    const path = join(messageBrokerDirectory, "base", generateFileName);
+    return [{ code: template, path }];
+  }
+
+  beforeUpdateDockerCompose(
+    dsgContext: DsgContext,
+    eventParams: CreateServerDockerComposeParams["before"]
+  ): CreateServerDockerComposeParams["before"] {
+    const KAFKA_NAME = "kafka";
+    const ZOOKEEPER_NAME = "zookeeper";
+    const NETWORK = "internal";
+    const ZOOKEEPER_PORT = "2181";
+    const KAFKA_PORT = "9092";
+    const newParams = merge(eventParams.updateProperties, {
+      services: {
+        [ZOOKEEPER_NAME]: {
+          image: "confluentinc/cp-zookeeper:5.2.4",
+          networks: [NETWORK],
+          environment: {
+            ZOOKEEPER_CLIENT_PORT: 2181,
+            ZOOKEEPER_TICK_TIME: 2000,
```

Comment:
we changes the props of this event - please sync with latest version and talk to @abrl91 if needed